src/sparseml/pytorch/image_classification/utils/helpers.py

```diff
@@ -450,3 +286,150 @@ def save_model_training(
                 info_lines.append(f"{loss}: {val_res.result_mean(loss).item()}")
 
         info_file.write("\n".join(info_lines))
+
+
+# device helpers
+def device_setup(model, rank: int, local_rank: int, device: Union[str, int]):
+    """
+    Move the model to the correct device.
+
+    :param model: The model to move to the correct device
+    :param rank: The rank of the process in network
+    :param local_rank: The rank of the process on the local machine
+    :param device: The device to move the model to
+    :return: A tuple of (model, device, ddp)
+    """
+    if rank == -1:
+        ddp = False
+    else:
+        torch.cuda.set_device(local_rank)
+        device = local_rank
+        ddp = True
+    model, device, _ = model_to_device(model=model, device=device, ddp=ddp)
+    return model, device, ddp
+
+
+# private methods
+def _get_collate_fn(arch_key: Optional[str] = None, task: Optional[Tasks] = None):
+    if not arch_key:
+        return None
+
+    is_ssd = "ssd" in arch_key.lower()
+    need_collate_function = (
+        is_ssd or "yolo" in arch_key.lower()
+    ) and task != Tasks.EXPORT
+    if need_collate_function:
+        return ssd_collate_fn if is_ssd else yolo_collate_fn
+    return None
+
+
+def _create_train_dataset_and_loader(
+    args: Any,
+    image_size: Tuple[int, ...],
+    task: Optional[Tasks] = None,
+) -> Tuple[Any, Any]:
+    need_train_data = not (
+        task == Tasks.EXPORT
+        or task == Tasks.PR_SENSITIVITY
+        and args.approximate
+        or task == Tasks.TRAIN
+        and args.eval_mode
+    )
+
+    if need_train_data:
+        with torch_distributed_zero_first(
+            args.local_rank,
+        ):  # only download once locally
+            train_dataset = DatasetRegistry.create(
+                args.dataset,
+                root=args.dataset_path,
+                train=True,
+                rand_trans=True,
+                image_size=image_size,
+                **args.dataset_kwargs,
+            )
+        sampler = (
+            torch.utils.data.distributed.DistributedSampler(train_dataset)
+            if args.rank != -1
+            else None
+        )
+        shuffle = True if sampler is None else False
+        batch_size = args.train_batch_size if task == Tasks.TRAIN else args.batch_size
+
+        arch_key = args.arch_key if hasattr(args, "arch_key") else None
+        train_loader = DataLoader(
+            train_dataset,
+            batch_size=batch_size,
+            shuffle=shuffle,
+            num_workers=args.loader_num_workers,
+            pin_memory=args.loader_pin_memory,
+            sampler=sampler,
+            collate_fn=_get_collate_fn(arch_key=arch_key, task=task),
+        )
+        print(f"created train_dataset: {train_dataset}")
+        return train_dataset, train_loader
+
+    return None, None
+
+
+def _create_val_dataset_and_loader(
```

Comment:
There is exactly the same function in `sparseml/integrations/pytorch/utils.py`. Why the duplicate?

Reply:
The same goes for some other functions e.g. save_model_training. If we are planning to retire `sparseml/integrations`, this is ok, otherwise, probably not.

Reply:
We are planning to retire `sparseml/integrations/pytorch` in favour of `sparseml.image_classification.*` cli utilities

Reply:
We are retiring `sparseml/integration/pytorch` in favor of `sparseml.image_classification.*` CLI utilities